eclair-node/src/main/scala/fr/acinq/eclair/api/JsonSerializers.scala

```diff
@@ -216,17 +302,9 @@ object JsonSupport extends Json4sSupport {
     new NodeAddressSerializer +
     new DirectionSerializer +
     new PaymentRequestSerializer +
-    new JavaUUIDSerializer
-
-  case class CustomTypeHints(custom: Map[Class[_], String]) extends TypeHints {
-    val reverse: Map[String, Class[_]] = custom.map(_.swap)
-
-    override val hints: List[Class[_]] = custom.keys.toList
-    override def hintFor(clazz: Class[_]): String = custom.getOrElse(clazz, {
-      throw new IllegalArgumentException(s"No type hint mapping found for $clazz")
-    })
-    override def classFor(hint: String): Option[Class[_]] = reverse.get(hint)
-  }
-
+    new JavaUUIDSerializer).withTypeHintFieldName("type") +
```

Comment:
Is it necessary to apply the `withTypeHintFieldName` to all the Formats? It seems that it's being applied to all serializers and not only the payment events, although i see that other classes aren't serialized with the `type` (perhaps because they're not in the list of custom type names) and it's a bit hard to find documentation for this method on json4s.

Reply:
Alternatively (but it's not strictly better) we could define two sets of formats in this way:
```
val formats = ... // default generic formats
val formatsWithTypeHint = formats.withTypeHintFieldName("type") + CustomTypeHint
```

Then use the latter only where necessary.

Reply:
From what I understand it will only have an effect for types for which you actually define a type hint.
I honestly don't see a reason to separate the two, having those type hints is beneficial everywhere (and the code is less messy if we don't have two separate formats).

Reply:
Yes that seems to be the effect, i left the comment because i wasn't sure of its behavior and i can't find any documentation about it.